File: scala/sdktest/CheckCardBlackListedEntityTest.scala
// Generated basic-flow test for the check_card_black_listed entity (model-driven;
// mirrors the java TestEntity generator). A dependency-free scala-cli test
// object driven by SdkEntityTestMain. Runs against the in-memory test
// transport seeded with the shipped CheckCardBlackListedTestData.json fixtures.

import java.util.{ArrayList, LinkedHashMap, List => JList, Map => JMap}

import voxgig.bluefintecsmerchantservicessdk.core.{Helpers, BluefinTecsMerchantServicesSDK}
import voxgig.bluefintecsmerchantservicessdk.utility.struct.Struct

object CheckCardBlackListedEntityTest {

  def run(rep: SdkTestReport): Unit = {
    rep.scope("check_card_black_listed.instance") {
      val testsdk = BluefinTecsMerchantServicesSDK.testSDK()
      val ent = testsdk.checkCardBlackListed(null)
      rep.check("check_card_black_listed.instance", ent != null, "expected non-null check_card_black_listed entity")
    }

    rep.scope("check_card_black_listed.basic") {
      val entityData = Helpers.toMapAny(SdkTestSupport.readJson(
          "../.sdk/test/entity/check_card_black_listed/CheckCardBlackListedTestData.json"))
      val options = new LinkedHashMap[String, Object]()
      options.put("entity", entityData.get("existing"))
      val client = BluefinTecsMerchantServicesSDK.testSDK(options, null)

      val idmap = new LinkedHashMap[String, Object]()
      idmap.put("check_card_black_listed01", "CHECK_CARD_BLACK_LISTED01")
      idmap.put("check_card_black_listed02", "CHECK_CARD_BLACK_LISTED02")
      idmap.put("check_card_black_listed03", "CHECK_CARD_BLACK_LISTED03")
      val now = System.currentTimeMillis()

      // CREATE
      val checkCardBlackListedRef01Ent = client.checkCardBlackListed(null)
      var checkCardBlackListedRef01Data = Helpers.toMapAny(Struct.getprop(
          Struct.getpath(entityData, "new.check_card_black_listed"), "check_card_black_listed_ref01"))
      val checkCardBlackListedRef01DataResult = checkCardBlackListedRef01Ent.create(checkCardBlackListedRef01Data, null)
      checkCardBlackListedRef01Data = Helpers.toMapAny(checkCardBlackListedRef01DataResult)
      rep.check("check_card_black_listed.create.map", checkCardBlackListedRef01Data != null, "expected create result to be a map")
    }
  }
}

File: dart/test/entity/check_card_black_listed/CheckCardBlackListedEntity_test.dart
// ignore_for_file: unused_import, unused_local_variable, non_constant_identifier_names

import 'dart:convert';
import 'dart:io';

import '../../harness.dart';
import '../../utility.dart';

import '../../../lib/BluefinTecsMerchantServicesSDK.dart';
import '../../../lib/utility/voxgig_struct.dart' as vs;

void tests() {
  describe('CheckCardBlackListedEntity', () {
    test('instance', (t) async {
      final testsdk = BluefinTecsMerchantServicesSDK.test();
      final ent = testsdk.CheckCardBlackListed();
      ok(null != ent);
    });



    test('basic', (t) async {

      final live = 'TRUE' == Platform.environment['BLUEFIN_TECS_MERCHANT_SERVICES_TEST_LIVE'];
      for (final op in ['create']) {
        if (maybeSkipControl(t, 'entityOp', 'check_card_black_listed.' + op, live)) {
          return;
        }
      }

      final setup = basicSetup();
      // The basic flow consumes synthetic IDs and field values from the
      // fixture (entity TestData.json). Those don't exist on the live API.
      // Skip live runs unless the user provided a real ENTID env override.
      if (true == setup['syntheticOnly']) {
        t.skip('live entity test uses synthetic IDs from fixture — set BLUEFIN_TECS_MERCHANT_SERVICES_TEST_CHECK_CARD_BLACK_LISTED_ENTID JSON to run live');
        return;
      }
      final client = setup['client'];
      final struct = setup['struct'];

      final isempty = struct.isempty;
      final select = struct.select;


      // CREATE
      final check_card_black_listed_ref01_ent = client.CheckCardBlackListed();
      dynamic check_card_black_listed_ref01_data = setup['data']['new']['check_card_black_listed']['check_card_black_listed_ref01'];

      check_card_black_listed_ref01_data = (await check_card_black_listed_ref01_ent.create(check_card_black_listed_ref01_data)).data();
      ok(null != check_card_black_listed_ref01_data);


    });
  });
}


Map<String, dynamic> basicSetup([dynamic extra]) {
  final options = <String, dynamic>{};

  final entityDataFile = resolveTestPath(
      '../.sdk/test/entity/check_card_black_listed/CheckCardBlackListedTestData.json');

  final entityDataSource = File(entityDataFile).readAsStringSync();

  final entityData = jsonDecode(entityDataSource);

  options['entity'] = entityData['existing'];

  var client = BluefinTecsMerchantServicesSDK.test(options, extra);
  final struct = client.utility().struct;
  final merge = struct.merge;
  final transform = struct.transform;

  dynamic idmap = transform(
      <dynamic>['check_card_black_listed01', 'check_card_black_listed02', 'check_card_black_listed03'],
      <String, dynamic>{
        '`\$PACK`': <dynamic>[
          '',
          <String, dynamic>{
            '`\$KEY`': '`\$COPY`',
            '`\$VAL`': <dynamic>['`\$FORMAT`', 'upper', '`\$COPY`'],
          }
        ]
      });

  // Detect whether the user provided a real ENTID JSON via env var. The
  // basic flow consumes synthetic IDs from the fixture file; without an
  // override those synthetic IDs reach the live API and 4xx. Surface this
  // to the test so it can skip rather than fail.
  final idmapEnvVal =
      Platform.environment['BLUEFIN_TECS_MERCHANT_SERVICES_TEST_CHECK_CARD_BLACK_LISTED_ENTID'];
  final idmapOverridden =
      null != idmapEnvVal && idmapEnvVal.trim().startsWith('{');

  final env = envOverride({
    'BLUEFIN_TECS_MERCHANT_SERVICES_TEST_CHECK_CARD_BLACK_LISTED_ENTID': idmap,
    'BLUEFIN_TECS_MERCHANT_SERVICES_TEST_LIVE': 'FALSE',
    'BLUEFIN_TECS_MERCHANT_SERVICES_TEST_EXPLAIN': 'FALSE',
    'BLUEFIN_TECS_MERCHANT_SERVICES_APIKEY': 'NONE',
  });

  idmap = env['BLUEFIN_TECS_MERCHANT_SERVICES_TEST_CHECK_CARD_BLACK_LISTED_ENTID'];

  final live = 'TRUE' == env['BLUEFIN_TECS_MERCHANT_SERVICES_TEST_LIVE'];

  if (live) {
    client = BluefinTecsMerchantServicesSDK(merge([
      <String, dynamic>{
        'apikey': env['BLUEFIN_TECS_MERCHANT_SERVICES_APIKEY'],
      },
      extra
    ]));
  }

  final setup = <String, dynamic>{
    'idmap': idmap,
    'env': env,
    'options': options,
    'client': client,
    'struct': struct,
    'data': entityData,
    'explain': 'TRUE' == env['BLUEFIN_TECS_MERCHANT_SERVICES_TEST_EXPLAIN'],
    'live': live,
    'syntheticOnly': live && !idmapOverridden,
    'now': DateTime.now().millisecondsSinceEpoch,
  };

  return setup;
}

